pkg/templates/templates.go

```diff
@@ -39,6 +42,7 @@ const (
 // - StopDelim customizes the stop delimiter used to distinguish a template action. This defaults
 // to "}}". If StartDelim is set, this must also be set.
 type Config struct {
```

Comment:
This config is not too big yet, so the approach here is fine. Just FYI, if it needs to grow more you could consider the approach outlined here: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

Reply:
That is pretty elegant in some ways but hard to understand for the API consumer. I think I prefer the Config struct approach but could be easily swayed.  :smile:

Reply:
I referred to passing optional arguments to functions  somewhere earlier in this review , I guess the technical term I was looking for was variadic function.. phew.

But I think what we have now works just  fine. We can fine tune as we go if needed.